### DISCUSSION TITLE
Split js provider library in a separate repo

### DISCUSSION THREAD

**Discussion by cammellos** [status-im/status-legacy#16624] [General]
In order to share code with desktop and better identify compatibility issues with other providers (metamask/ether.js) we should separate the js code in a separate repo and share it with desktop.

**Comment by iurimatias** [comment]
On that note, wondering if this also makes sense for other shared stuff like for e.g, translations, and assets - so long it doesn't introduce a noticeable overhead

**Comment by flexsurfer** [comment]
I've looked into this and i should say, there will be not much profit from separating provider.js , because most logic is in status-react and status-go, provider.js just a tiny warper used for redirect request from the webview to the application, also we can't use it as npm library, it should be included in the project as a string, so i wouldn't separate it

**Comment by cammellos** [comment]
I think it's still worth it, 
currently some dapps work on desktop and some don't work on mobile, having it in a separate repo will help have a consistent experience between the two projects, leverage shared resources and any fix that we implement will benefits both.

Also why do you say we can't include it as an npm library? 
We can slurp the file after pulling it right?

**Comment by flexsurfer** [comment]
yeah probably we can slurp it from node modules folder

**Comment by cammellos** [comment]
Moving to a discussion, good to re-evaluate as we start working on brwoser
